tests/testPhotoCal.py

```diff
@@ -149,9 +149,42 @@ def testFlags(self):
         self.assertLessEqual(used, candidates)
         self.assertEqual(reserved, 0)
 
+    def testRandom(self):
+        """test that setting expId makes some difference in which are reserved
+           but that the total count of reserved candidates is the same."""
+        schema = self.srcCat.schema
+        task = PhotoCalTask(self.refObjLoader, config=self.config, schema=schema)
+        mapper = afwTable.SchemaMapper(self.srcCat.schema, schema)
+
         #   set the reserve fraction, and see if the right proportion are reserved.
+        cat = afwTable.SourceCatalog(schema)
+        for name in self.srcCat.schema.getNames():
+            mapper.addMapping(self.srcCat.schema.find(name).key)
+        cat.extend(self.srcCat, mapper=mapper)
+        reservedSum = 0
         self.config.reserveFraction = .3
-        task.run(exposure=self.exposure, sourceCat=cat)
+        task.run(exposure=self.exposure, sourceCat=cat, expId=12345)
+        candidates = 0
+        reserved = 0
+        used = 0
+        for source in cat:
+            if source.get("calib_photometryCandidate"):
+                candidates += 1
+            if source.get("calib_photometryUsed"):
+                used += 1
+            if source.get("calib_photometryReserved"):
+                reservedSum += source.getId()
+                reserved += 1
+        self.assertEqual(reserved, int(.3 * (candidates + reserved)))
+        self.assertLessEqual(used, (candidates - reserved))
```

Comment:
None of these asserts (here and below) are actually meaningful unless you have confirmed that `used`, `candidates`, and `reserved` are all `> 0`